app/state.go

```diff
@@ -589,12 +589,15 @@ func (uc *updateAfterFirstCommitState) Handle(ctx *StateContext, c Controller) (
 
 	var firstErr error
 
-	for _, i := range c.GetInstallers()[1:] {
-		err := i.CommitUpdate()
-		if err != nil {
-			log.Errorf("Error committing %s payload: %s", i.GetType(), err.Error())
-			if firstErr == nil {
-				firstErr = err
+	installers := c.GetInstallers()
+	if len(installers) > 1 {
```

Comment:
Q: More of a fix, than new functionality, am I right?

Reply:
Kind of.

Before `GetInstallers()` it was guaranteed to have at least one element, so `GetInstallers()[1:]` would just be an empty slice. However now it can have zero elements, so `[1:]` just crashes badly.